src/kudu/util/thread.h

```diff
@@ -368,6 +368,47 @@ class Thread : public RefCountedThreadSafe<Thread> {
 // the given entity. If 'web' is NULL, does not register the path handler.
 Status StartThreadInstrumentation(const scoped_refptr<MetricEntity>& server_metrics,
                                   WebCallbackRegistry* web);
+
+// Container class for any details we want to capture about a thread
+// TODO: Add start-time.
+// TODO: Track fragment ID.
+// fb: Move to header file, so can be used by plugin
+class ThreadDescriptor {
+  public:
+  ThreadDescriptor() { }
+  ThreadDescriptor(std::string category, std::string name, int64_t thread_id)
+      : name_(std::move(name)),
+        category_(std::move(category)),
+        thread_id_(thread_id) {}
+
+  const std::string& name() const { return name_; }
+  const std::string& category() const { return category_; }
+  int64_t thread_id() const { return thread_id_; }
+  int priority() const { return priority_; }
+  void setPriority(int p) { priority_ = p; }
+
+  private:
+  std::string name_;
+  std::string category_;
```

Comment:
Lets have a one line description for what these fields are (especially `category_`)